linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/manager/am/selector/rule/AvailableNodeSelectRule.scala

```diff
@@ -34,13 +34,15 @@ class AvailableNodeSelectRule extends NodeSelectRule with Logging{
     if (null != nodes) {
       nodes.filter {
         case amNode: AMNode =>
-          if (! NodeStatus.isLocked(amNode.getNodeStatus) && NodeStatus.isAvailable(amNode.getNodeStatus) ) {
+          if (!NodeStatus.isLocked(amNode.getNodeStatus)
+            && NodeStatus.isAvailable(amNode.getNodeStatus)
+            && NodeStatus.isEngineNodeHealthy(amNode.getNodeStatus).equals(NodeHealthy.Healthy)) {
```

Comment:
Hello, nodeHealthy should be used for judgment here, and status cannot be used to judge nodeHealthy